도미노 충돌 구현 및 게임 판정 구현

jinsu421 opened the pull request:
게임을 위한 매니저 코드들 추가.
공과 도미노 충돌 및 기초 카메카 추가로 테스트 환경 구현.
-> 이후 ObjectManager 코드에서 도미노를 생성하도록 추가변경하도록 하겠습니다.
README 파일에도 변경점 추가
위의 4가지 항목들을 변경하였습니다. ╰(*°▽°*)╯

SWgil commented on JINSU/Domino/.vscode/extensions.json:
```diff
@@ -0,0 +1,5 @@
+{
```
.vscode도 .gitignore에 추가하고, pr에서 제외해주세요 😊

jinsu421 replied:
gitignore에 vscode도 추가하였습니다 😎

SWgil commented on JINSU/Domino/Assets/CodeScript/ChasingCamera.cs:
```diff
@@ -0,0 +1,62 @@
+using System.Resources;
+using UnityEngine;
```
벌써 카메라 기능까지 구현 좋습니다 👍👍
카메라 관련 기능을 별도의 커밋으로 분리해서 올려주시면 리뷰하기 쉬울 것 같습니다

jinsu421 replied:
커밋 분리 이슈는 당사자와 협의 후 적용하였습니다.
카메라는 추후에 변경 할 예정입니다.

SWgil commented on JINSU/Domino/Assets/CodeScript/ChasingCamera.cs:
```diff
@@ -0,0 +1,62 @@
+public class ChasingCamera : MonoBehaviour
+{
+
+    // Start is called once before the first execution of Update after the MonoBehaviour is created
```
기본 주석은 지우는 걸로 하는게 어떨까요?

SWgil commented on JINSU/Domino/Assets/CodeScript/ChasingCamera.cs:
```diff
@@ -0,0 +1,62 @@
+// for testing
+    private void MoveCamera()
+    {
+        float moveZ = 0f;
+        float moveX = 0f;
+        if (Input.GetKey(KeyCode.W))
+        {
+            moveZ += 1f;
+        }
+
+        if (Input.GetKey(KeyCode.S))
+        {
+            moveZ -= 1f;
+        }
+
+        if (Input.GetKey(KeyCode.A))
+        {
+            moveX -= 1f;
+        }
+
+        if (Input.GetKey(KeyCode.D))
+        {
+            moveX += 1f;
+        }
+
+        transform.Translate(new Vector3(moveX, 0f, moveZ) * 0.1f);
```
직접 wsad에 대해서 if문을 처리하기 보다, 유니티의 Input Manager를 사용하는 방식으로 변경해주면 더 간결하게 작성할 수 있을 다고 하네요
https://docs.unity3d.com/kr/2021.3/Manual/class-InputManager.html
```suggestion
      float moveSpeed = 10;
//Define the speed at which the object moves.

float horizontalInput = Input.GetAxis("Horizontal");
//Get the value of the Horizontal input axis.

float verticalInput = Input.GetAxis("Vertical");
//Get the value of the Vertical input axis.

transform.Translate(new Vector3(horizontalInput, verticalInput, 0) * moveSpeed * Time.deltaTime);
//Move the object to XYZ coordinates defined as horizontalInput, 0, and verticalInput respectively.
```

jinsu421 replied:
다음주에 작업해보도록 하겠습니다. ✌️

SWgil commented on JINSU/Domino/Assets/CodeScript/ChasingCamera.cs:
```diff
@@ -0,0 +1,62 @@
+
+    /*
+    private void Update() {
+    // ���� ������Ʈ�� �������� ���� �ӵ��� ���� �̵��ϴ� ó��
```
TODO를 같이 적어주면 추후에 놓치지않고 관리하기 좋을 것 같습니다 😊

SWgil commented on JINSU/Domino/Assets/CodeScript/GameManager.cs:
```diff
@@ -0,0 +1,132 @@
+using System.Collections.Generic;
+
+
+//Courtine�� ���� ���������� (prviate,public,protected�� �����������)
```
무슨 의미의 주석인지 설명 해주실 수 있으실까요?

jinsu421 replied:
초반에 공부용으로 썻던 주석용입니다
![CalculatingPuzzledGIF](https://github.com/user-attachments/assets/a91f035d-108e-4ea1-a136-8c6cdd624cdd)
.

SWgil commented on JINSU/Domino/Assets/CodeScript/GameManager.cs:
```diff
@@ -0,0 +1,132 @@
+
+        if (isGamePlaying == true )
+        {
+            // LoadMenuScene or LoadOptionScene
```
여기도 TODO 붙혀주세요

SWgil commented on JINSU/Domino/Assets/CodeScript/InvisibleBall.cs:
```diff
@@ -0,0 +1,34 @@
+
+        if (Input.GetKey(KeyCode.K))
+        {
+            GetComponent<Rigidbody>().AddForce(speed * -1,0, 0,ForceMode.Impulse);
```
왜 `ForceMode.Impulse`를 사용하신건지 문의드립니다

jinsu421 replied:
정지->이동 상태로 힘을 가해줘야 했기때문에,
 현실 물리상황(중력 등)을 고려한 상황에선 Impulse가 낫다고 판단하였습니다.
참고 사이트 올립니다 https://blog.naver.com/gold_metal/221486016593

SWgil commented on JINSU/Domino/Assets/CodeScript/InvisibleBall.cs:
```diff
@@ -0,0 +1,34 @@
+    {
+        if (collisionInfo.gameObject.CompareTag("DominoBlock"))
+        {
+            Debug.Log(string.Format("{0} ������Ʈ�� {1} �� �浹�߽��ϴ�.", gameObject.name, collisionInfo.gameObject.name));
```
👍👍

SWgil commented on JINSU/Domino/Assets/CodeScript/InvisibleBall.cs:
```diff
@@ -0,0 +1,34 @@
+
+    void OnCollisionEnter(Collision collisionInfo)
+    {
+        if (collisionInfo.gameObject.CompareTag("DominoBlock"))
```
추후에 값을 "DominoBlock" 스트링 말고, 존재하는 TAG 리스트에서 가져올 수 있다던가 하면, 실수를 줄일 수 있을 것 같습니다 (가능하다면)

jinsu421 replied:
추후 오브젝트매니저를 작업하게 된다면, 해당 코드에서 충돌감지 여부를 판독하도록 해결해보겠습니다. 👌

SWgil commented on JINSU/Domino/Assets/CodeScript/Domino.cs:
```diff
@@ -0,0 +1,17 @@
+    // Start is called once before the first execution of Update after the MonoBehaviour is created
+    void Start()
+    {
+
```
미리 스크립트를 작성한다면 TODO로 의도를 적어주면, 리뷰하는 입장에서 실수인지 판단할 수 있을 것 같습니다

jinsu421 replied:
기존 코드에 자동생성된 주석코드와 TODO 관련 이슈를 한번에 정리하였습니다.

SWgil commented on JINSU/Domino/Assets/Scenes/SampleScene.unity:
```diff
@@ -38,12 +38,12 @@ RenderSettings:
   m_ReflectionIntensity: 1
   m_CustomReflection: {fileID: 0}
   m_Sun: {fileID: 0}
-  m_IndirectSpecularColor: {r: 0.18028378, g: 0.22571412, b: 0.30692285, a: 1}
```
SampleScene을 지우는게 나을 것 같습니다

SWgil commented on JINSU/Domino/ProjectSettings/TagManager.asset:
```diff
@@ -2,8 +2,10 @@
-  tags: []
+  serializedVersion: 3
+  tags:
+  - UnseeingBall
```
이 태그도 미리 만들어둔건가요?

jinsu421 replied:
충돌 구현에 필요해서 태그를 사용하였습니다만, 추후 변경할 예정입니다.

jinsu421 committed:
feat 게임용 매니저들 추가 및 도미노 충돌 구현

1.GameManager, ObjectManager(빈코드), EffectManager(빈코드)로 추가
2.도미노 및 공 생성으로 충돌 구현
하였습니다.

jinsu421 committed:
feat 카메라 코드 추가

도미노를 따라다니는 카메라를 위한 코드 추가(지금은 테스트용)

jinsu421 committed:
chore vscode 개발환경으로변경 및 커멘트에 따른 변경

1.unity 기본 생성 주석 삭제
2.샘플씬 삭제
3.vscode로 변경하면서 이슈로 적은 개발환경으로 셋팅

SWgil reviewed (approved):
LGTM
